controllers/certs.go

```diff
@@ -87,7 +88,16 @@ func (r *EtcdadmClusterReconciler) generateCAandClientCertSecrets(ctx context.Co
 		Generated: true,
 	}
 	s := apiServerClientCertKeyPair.AsSecret(client.ObjectKey{Name: cluster.Name, Namespace: cluster.Namespace}, *metav1.NewControllerRef(etcdCluster, etcdv1.GroupVersion.WithKind("EtcdadmCluster")))
-	if err := r.Client.Create(ctx, s); err != nil && !apierrors.IsAlreadyExists(err) {
+	secretToPatch := s.DeepCopy()
```

Comment:
i dont quite understand why we need to have a deep copy of the secret

Reply:
So basically its for how `CreateOrPatch` works. If the object is found, its expected that the changes needed should be redone in a call back func. 

So on line 98 `s.Data = secretToPatch.Data` i am updating the secret data again using the deep copied object.